lru.py

```diff
@@ -0,0 +1,238 @@
+""" LRU(size, callback=None) -> new LRU dict that can store up to size elements
+    An LRU dict behaves like a standard dict, except that it stores only fixed
+    set of elements. Once the size overflows, it evicts least recently used
+    items.  If a callback is set it will call the callback with the evicted key
+     and item.
+
+    Eg:
+    >>> l = LRU(3)
+    >>> for i in range(5):
+    >>>   l[i] = str(i)
+    >>> l.keys()
+    [2,3,4]
+
+    Note: An LRU(n) can be thought of as a dict that will have the most
+    recently accessed n items.
+"""
+
+class Node(object):
```

Comment:
Maybe consider using `__slots__` on Node objects to avoid wasting memory on their dicts?